arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceOptionsViewController.swift

```diff
@@ -0,0 +1,258 @@
+            }
+        }
+        dismiss(animated: true)
+    }
+    
+    @IBAction func cancelBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        dismiss(animated: true)
+    }
+    
+    // MARK: UI and data binding methods
+    
+    /// Convert the values to matching data types.
+    ///
+    /// - Parameters:
+    ///   - value: The right hand side value used in the conditional expression.
+    ///   - dataType: An `AGSUtilityNetworkAttributeDataType` enum case.
+    ///
+    /// - Note: The input value can either be an `AGSCodedValue` populated from the left hand side
+    ///         attribute's domain, or a numeric value entered by the user.
+    ///
+    /// - Returns: Converted value.
+    func convertToDataType(value: Any, dataType: AGSUtilityNetworkAttributeDataType) -> Any {
+        switch dataType {
+        case .integer:
+            return value as! Int64
+        case .float:
+            return value as! Float
+        case .double:
+            return value as! Double
+        case .boolean:
+            return value as! Bool
+        default:
+            return value
+        }
+    }
+    
+    func updateCellStates() {
+        // Disable the value cell when attribute is unspecified.
+        if let selectedAttribute = selectedAttribute {
+            if selectedAttribute.domain is AGSCodedValueDomain {
+                // Indicate that a new view controller will display.
+                valueCell.accessoryType = .disclosureIndicator
+            } else {
+                // Indicate that an alert will show.
+                valueCell.accessoryType = .none
+            }
+            valueCell.textLabel?.isEnabled = true
+            valueCell.isUserInteractionEnabled = true
+        } else {
+            // Enable the value cell when an attribute is specified.
+            valueCell.textLabel?.isEnabled = false
+            valueCell.isUserInteractionEnabled = false
+        }
+    }
+    
+    // Transition to the attribute options view controller.
+    func showAttributePicker() {
+        let selectedIndex = possibleAttributes.firstIndex { $0 == selectedAttribute } ?? -1
+        let optionsViewController = OptionsTableViewController(labels: possibleAttributes.map { $0.name }, selectedIndex: selectedIndex) { newIndex in
+            self.selectedAttribute = self.possibleAttributes[newIndex]
+            self.navigationController?.popViewController(animated: true)
+        }
+        optionsViewController.title = "Attributes"
+        show(optionsViewController, sender: self)
+    }
+    
+    // Transition to the comparison options view controller.
+    func showComparisonPicker() {
+        let selectedIndex = selectedComparison?.rawValue ?? -1
+        // An array of `AGSUtilityAttributeComparisonOperator` init from their raw value.
+        let attributeComparisonOperators = (0...9).map { AGSUtilityAttributeComparisonOperator(rawValue: $0)! }
```

Comment:
We discussed it earlier and I decided to make this change. Now I'm not satisfied about it. 😩 
I'll make some tweaks tomorrow.